### DISCUSSION TITLE
Ending without left barline

### DISCUSSION THREAD

**Discussion by hbitteur** [w3c-cg/musicxml#443] [Q&A] [answered]
Using Audiveris OMR, I stumbled on a score image with a system start as follows:

![image](https://user-images.githubusercontent.com/22053634/134802693-b99bfaea-6bea-41f0-b44e-6336751e7e35.png)

How should Audiveris export the 'ending' element to MusicXML format? 

According to MusicXML reference documentation (4.0), 'ending' element is a child of 'barline' element.
The problem in this example is there is no barline on the left side of the containing measure.

Audiveris, as an OMR software, is very "image-oriented". It can't see any barline when there is none.
Should I force it to imagine there is a kind of "virtual barline" there, just to be able to export the ending sign?
I hate doing this :-)

Thanks for any help
/Hervé (Audiveris owner)

**Comment by mscuthbert** [comment]
Ah, interesting -- there's no way of knowing from this staff itself whether this is a mid-measure start of the volta or a start of measure volta -- you have to look at the end of the previous staff to see if there is a barline there, and then the volta is rendered "on" that barline.  

From https://www.w3.org/2021/06/musicxml40/musicxml-reference/elements/ending/ I think that the best representation of this is an `<ending>` attached to a `<barline location="left"><bar-style>none</bar-style></barline>`

(p.s. nice to see you here Hervé -- sorry we haven't crossed paths at conferences in years!)

**Comment by hbitteur** [comment]
Hi Michael,

![image](https://user-images.githubusercontent.com/22053634/134912170-75803457-f8c8-4827-8a4d-461b2bdddb9f.png)

Using this larger view, we can see that this is a start of measure volta rather than a mid-measure.

So, we are back to deal with a virtual barline on the left side of our measure. 
And you are right, at [https://www.w3.org/2021/06/musicxml40/musicxml-reference/data-types/bar-style/](https://www.w3.org/2021/06/musicxml40/musicxml-reference/data-types/bar-style/), documentation on  "none" barline style says: "No barline appears".

This is not comfortable for an OMR, which here won't be able to use the presence of a barline interpretation to support the ending element interpretation. I think I can modify the algorithm to accept this configuration at the very beginning af a staff without concrete left barline. It's worth being tried.

P.S. I'm retired now and don't have the opportunity to attend "physical" conferences, especially now that MuseScore initial owners have left. But I'm more active than ever on Audiveris evolution...

**Comment by mdgood** [comment]
I would not recommend using the `<bar-style>` element in this situation. That is because a value of none would make the barline invisible even if it moved mid-system. I think it would be best for scanning software to treat endings beginning at the start of a system as a special case.

**Comment by hbitteur** [comment]
@mdgood
Since the <ending> element needs to be nested within a <barline> element, we need to create such (artificial) barline element from nothing.
If I understand your remark correctly, we should avoid the 'none' bar-style, and rather pick up some other bar-style.
The barline would thus stay visible to prevent any unwanted user edition.
Is this interpretation the correct one?

**Comment by mdgood** [comment, marked as answer]
@hbitteur The `<bar-style>` element is optional so just leave it out. That's what usually happens with `<ending>` elements - the `<barline>` element is just the place where the `<ending>` element goes. See the [`<ending>` example](https://www.w3.org/2021/06/musicxml40/musicxml-reference/examples/ending-element/) for instance.
